src/pages/task/testsTable/TestsTableCore.tsx

```diff
@@ -165,30 +81,61 @@ export const TestsTableCore: React.FC<ValidInitialQueryParams> = ({
       });
     }
   }, [networkStatus]);
-  if (
-    (sort !== prevSort || category !== prevCategory) &&
-    networkStatus === NetworkStatus.ready &&
-    !error
-  ) {
-    fetchMore({
-      variables: {
-        cat: category,
-        dir: sort === SortQueryParam.Asc ? "ASC" : "DESC",
-        pageNum: 0,
-        limitNum: LIMIT
-      },
-      updateQuery: (
-        prev: UpdateQueryArg,
-        { fetchMoreResult }: { fetchMoreResult: UpdateQueryArg }
-      ) => {
-        if (!fetchMoreResult) {
-          return prev;
+
+  // this fetch is when url params change (sort direction, sort category, status list)
+  // and the page num is set to 0
+  useEffect(() => {
+    if (
+      (sort !== prevSort ||
+        category !== prevCategory ||
+        statuses !== prevStatuses) &&
+      networkStatus === NetworkStatus.ready &&
+      !error
```

Comment:
what is the value-add of checking if url are not equal to previous value after a url param change is triggered?

Reply:
the dependency list for the useEffect has many variables aside from url params such as network status and error. This logic is necessary to make sure we make a request only when the relevant query params have changed.

Reply:
instead you could remove this logic, which is a difficult to read, and replace it with `history.listen` which will attach an event listener to the url. within the cb you can call fetchMore which will only be called when the url changes.

Reply:
https://reacttraining.com/react-router/web/api/Hooks/usehistory

Reply:
`listen` seems good to use but I still think it's meaningful to only dispatch the request when a relevant parameter has changed. The callback passed to `listen` has a location argument that we can deconstruct to examine the query parameters but there isn't a way to see which query parameters have changed from the previous render.

Reply:
> I still think it's meaningful to only dispatch the request when a relevant parameter has changed

when will a parameter change without it being relevant?

>  I would still want to compare the previous query param to the current

when will a parameter change where it is the same as before?

Reply:
In the future we may have other query param in the URL alongside the query params that control the task table.

Reply:
that sounds hypothetical though and therefore an equal argument is that we could never have irrelevant query params that control the task table. 

right now all params are relevant and updating on change will have no negative performance consequences.